client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelInitializer.java

```diff
@@ -175,14 +179,21 @@ private void initSsl(Channel channel) {
       } else {
          sslContext = new JdkSslContext(ssl.sslContext(), true, ClientAuth.NONE);
       }
+
       SslHandler sslHandler = sslContext.newHandler(channel.alloc(), ssl.sniHostName(), -1);
-      if (ssl.sniHostName() != null) {
-         SSLParameters sslParameters = sslHandler.engine().getSSLParameters();
-         sslParameters.setServerNames(Collections.singletonList(new SNIHostName(ssl.sniHostName())));
-         sslHandler.engine().setSSLParameters(sslParameters);
+      String sniHostName;
+      if (cluster != null && cluster.getName() != null) {
+         sniHostName = cluster.getName();
```

Comment:
shouldn't be the `sniHostName` instead of the cluster name??

Reply:
Done